src/debug-generator/debug-generator.c

```diff
@@ -88,6 +109,23 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
 
                 return free_and_strdup_warn(&arg_default_unit, value);
 
+        } else if (proc_cmdline_key_streq(key, "systemd.boot_breakpoint")) {
+
+                if (value) {
+                        arg_boot_breakpoint = boot_breakpoint_type_from_string(value);
+                        if (arg_boot_breakpoint < 0)
+                                return arg_boot_breakpoint;
+                        else if (!in_initrd() && IN_SET(arg_boot_breakpoint,
+                                                        BOOT_BREAKPOINT_PRE_SYSROOT_MOUNT,
+                                                        BOOT_BREAKPOINT_PRE_SWITCH_ROOT)) {
+                                log_warning("Breakpoint '%s' only valid in the initrd, ignoring.", value);
+                                arg_boot_breakpoint = _BOOT_BREAKPOINT_TYPE_INVALID;
+                        }
+                } else if (in_initrd())
+                        arg_boot_breakpoint = BOOT_BREAKPOINT_PRE_SWITCH_ROOT;
```

Comment:
We should log if we're ignoring the option when specified without a value when not in the initrd.